### PR TITLE
Rename: `show caches` -> `show filesystem caches`, `describe cache` -> `describe filesystem cache`

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Rename cache commands: `show caches` -> `show filesystem caches`, `describe cache` -> `describe filesystem cache`

### CODE REVIEW

File: docs/en/operations/storing-data.md [hunk 1]
```diff
@@ -213,9 +213,9 @@ Cache **commands**:
 
 - `SYSTEM DROP FILESYSTEM CACHE (<path>) (ON CLUSTER)`
 
-- `SHOW CACHES` -- show list of caches which were configured on the server.
+- `SHOW FILESYSTEM CACHES` -- show list of caches which were configured on the server. (For versions <= `22.8` commands is named `SHOW CACHES`)
```
Comment (rschu1ze) [inline docs/en/operations/storing-data.md:216]: In my local CH, `SHOW CACHES` returns an empty result so I can't reakkt tell what it *should* return. Some remarks:

- `SHOW CACHES` / `SHOW FILESYSTEM CACHES` should be documented on https://clickhouse.com/docs/en/sql-reference/statements/show

- Likewise, https://clickhouse.com/docs/en/operations/caches/ could also mention the filesystem cache.

- If I read this PR correctly, the previous statement `SHOW CACHES` *only* referred to the filesystem cache(s). In that case, the previous naming was indeed misleading. Just wondering if there is a `SHOW` command for other cache types, e.g. the regexp cache?

- About documentation again. L. 214 is about dropping the filesystem cache. Could you please add that also to https://clickhouse.com/docs/en/sql-reference/statements/system?

- L. 216 should say "show list of filesystem caches [...]" instead of "show list of caches".

- Minor: (l. 216) "commands is names" --> "the command is named"

- About l. 218, it would maybe be cool to document DESCRIBE CACHE in https://clickhouse.com/docs/en/sql-reference/statements/describe-table.

File: src/Storages/NATS/NATSSettings.h [hunk 2]
```diff
@@ -24,7 +24,8 @@ class ASTStorage;
     M(Milliseconds, nats_flush_interval_ms, 0, "Timeout for flushing data from NATS.", 0) \
     M(String, nats_username, "", "NATS username", 0) \
     M(String, nats_password, "", "NATS password", 0) \
-    M(String, nats_token, "", "NATS token", 0)
+    M(String, nats_token, "", "NATS token", 0) \
+    M(UInt64, nats_startup_connect_tries, 5, "Number of connect tries at startup", 0) \
```
Comment (rschu1ze) [inline src/Storages/NATS/NATSSettings.h:28]: Unrelated?

File: src/Storages/NATS/StorageNATS.cpp [hunk 3]
```diff
@@ -92,10 +92,24 @@ StorageNATS::StorageNATS(
 
     try
     {
-        connection = std::make_shared<NATSConnectionManager>(configuration, log);
-        if (!connection->connect())
-            throw Exception(ErrorCodes::CANNOT_CONNECT_NATS, "Cannot connect to {}. Nats last error: {}",
-                            connection->connectionInfoForLog(), nats_GetLastError(nullptr));
+        size_t num_tries = nats_settings->nats_startup_connect_tries;
```
Comment (rschu1ze) [inline src/Storages/NATS/StorageNATS.cpp:95]: Unrelated?

File: tests/queries/0_stateless/02344_show_caches.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+-- Tags: no-fasttest
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02344_show_caches.sql:1]: The test runs in fractions of a millisecond, no reason to mark it "no-fasttest". Same for "02344_describe_cache". (just saying ... it has been like that before of course)

**Review by rschu1ze — APPROVED** [review]

File: src/Storages/NATS/NATSSettings.h (hunk 2 above)
Comment (kssenii) [reply src/Storages/NATS/NATSSettings.h:28]: yes, thanks, committed by accident

File: docs/en/operations/storing-data.md (hunk 1 above)
Comment (kssenii) [reply docs/en/operations/storing-data.md:216]: btw not sure if it these cache commands should be documented an show/, describe/, system/ sections, because they are documented on the page devoted to filesystem caches and there is a description for all of them. As a result there will be 2 places to update and keep in sync, as a result they might diverge...

> Likewise, https://clickhouse.com/docs/en/operations/caches/ could also mention the filesystem cache.

it already does...

> Just wondering if there is a SHOW command for other cache types, e.g. the regexp cache?

No, there isn't.

File: tests/queries/0_stateless/02344_show_caches.sql (hunk 4 above)
Comment (kssenii) [reply tests/queries/0_stateless/02344_show_caches.sql:1]: No, actually there is a reason - there are no caches in fast test, so in fast test there will be a different .reference file

Comment (rschu1ze) [reply tests/queries/0_stateless/02344_show_caches.sql:1]: Interesting, didn't know that.

File: docs/en/operations/storing-data.md [hunk 5]
```diff
@@ -213,10 +213,32 @@ Cache **commands**:
 
 - `SYSTEM DROP FILESYSTEM CACHE (<path>) (ON CLUSTER)`
 
-- `SHOW FILESYSTEM CACHES` -- show list of caches which were configured on the server. (For versions <= `22.8` commands is named `SHOW CACHES`)
+- `SHOW FILESYSTEM CACHES` -- show list of filesystem caches which were configured on the server. (For versions <= `22.8` commands is named `SHOW CACHES`)
```
Comment (rschu1ze) [inline docs/en/operations/storing-data.md:216]: (Minor: "commands" --> "command" or "the command")

File: docs/en/operations/storing-data.md (hunk 1 above)
Comment (rschu1ze) [reply docs/en/operations/storing-data.md]: Yepp. Having a full description in just a single place is a good idea.

**Review by rschu1ze — APPROVED** [review]

**Review by rschu1ze — APPROVED** [review]
